ClangTidy.cmake

```
@@ -229,6 +265,11 @@ AnalyzeTemporaryDtors: true
   # Only lint targets created in this repository. Later on we will create 2 targets: clang-tidy-all will lint all "core" targets, executables and libraries clang-tidy-world will
   # lint everything including test suites
   swift_list_compilable_targets(all_targets ONLY_THIS_REPO SWIFT_TYPES "executable" "library")
+  if(x_WITHOUT_SWIFT_TYPES)
+    swift_list_compilable_targets(all_targets_without_swift_types ONLY_THIS_REPO)
+    list(APPEND all_targets ${all_targets_without_swift_types})
+    list(REMOVE_DUPLICATES all_targets)
+  endif()
```

Might try this as it's a little more concise. 

```suggestion
  if (x_WITHOUT_SWIFT_TYPES)
    swift_list_compilable_targets(all_targets ONLY_THIS_REPO TYPES "EXECUTABLE" "STATIC_LIBRARY" "SHARED_LIBRARY")
  else()
    swift_list_compilable_targets(all_targets ONLY_THIS_REPO SWIFT_TYPES "executable" "library")
  endif()
```

---

@krisukox looking at this again are we picking up test files now? Maybe it was better how we had it before. Probably want to exclude test files from linting.